creational/singleton.md

```diff
@@ -16,9 +16,11 @@ var (
 )
 
 func New() singleton {
-	once.Do(func() {
-		instance = make(singleton)
-	})
+	if instance == nil {
+		once.Do(func() {
+			instance = make(singleton)
+		})
+	}
```

Comment:
As the documentation of `sync` describes, https://golang.org/src/sync/once.go?s=1473:1500#L30

```
// Do calls the function f if and only if Do is being called for the
// first time for this instance of Once. In other words, given
// 	var once Once
// if once.Do(f) is called multiple times, only the first call will invoke f,
// even if f has a different value in each invocation. A new instance of
// Once is required for each function to execute.
//
// Do is intended for initialization that must be run exactly once. Since f
// is niladic, it may be necessary to use a function literal to capture the
// arguments to a function to be invoked by Do:
// 	config.once.Do(func() { config.init(filename) })
```

Reply:
ohh of course, thanks for the clarification, i will close the PR